plugins/ccgx/fu-ccgx-hpi.c

```diff
@@ -0,0 +1,1633 @@
+ *
+ * Returns: %TRUE for success
+*/
+gboolean
+fu_ccgx_hpi_cmd_get_device_data (FuDevice *device,
+				 CyHPIHandle *hpi_handle,
+				 PDDeviceData *device_data,
+				 GError **error)
+{
+	guint32 device_versions[6] = {0};
+	guint8 device_mode = 0;
+	guint16 silicon_id = 0;
+	CCGxPartInfo *ccgx_info = NULL;
+	g_autofree guint8 *row_buffer = NULL;
+	guint32 row_size = 0;
+	guint32 row_max = 0;
+	guint32 fw_meta_row_num;
+	guint32 fw_meta_offset = 0;
+	guint8 fw_index = 0;
+	CCGxMetaData *metadata = NULL;
+	PDFWAppVersion *fw_version = NULL;
+
+	g_return_val_if_fail (hpi_handle != NULL, FALSE);
+	g_return_val_if_fail (device_data != NULL, FALSE);
+
+	if (!fu_ccgx_hpi_get_device_mode (device,hpi_handle,&device_mode,error)) {
+		return FALSE;
+	}
+
+	device_data->fw_mode = (FWMode)(device_mode & 0x03);
+
+	if (!fu_ccgx_hpi_get_silicon_id (device,hpi_handle,&silicon_id,error)) {
+		return FALSE;
+	}
+
+	device_data->silicon_id = (guint16)silicon_id;
+
+	device_data->current_version.val = 0;
+	if (device_data->fw_mode != FW_MODE_BOOT) {
+		if (!fu_ccgx_hpi_get_device_version (device, hpi_handle, device_versions, error)) {
+			return FALSE;
+		}
+		device_data->fw_version[FW_MODE_FW1].val = device_versions[3];
+		device_data->fw_version[FW_MODE_FW2].val = device_versions[5];
+		device_data->current_version.val = device_data->fw_version[device_data->fw_mode].val ;
+	} else {
+		g_warning("device in boot mode");
+	}
+
+	ccgx_info = fu_ccgx_util_find_ccgx_info (silicon_id);
+	if (ccgx_info == NULL) {
+		g_set_error (error,
+			     FWUPD_ERROR,
+			     FWUPD_ERROR_NOT_SUPPORTED,
+			     "not supported Silicon ID");
+		return FALSE;
+	}
+
+	row_size = ccgx_info->flash_row_size;
+	device_data->fw_row_size = row_size;
+
+	if (row_size > CYACD_FLASH_ROW_MAX) {
+		g_set_error (error,
+			     FWUPD_ERROR,
+			     FWUPD_ERROR_NOT_SUPPORTED,
+			     "Not row size");
+		return FALSE;
+	}
+
+	row_max = ccgx_info->flash_size / row_size;
+	row_buffer = g_malloc0(row_size);
+
+	if (row_size == 128) {
+		fw_meta_offset = HPI_META_DATA_OFFSET_ROW_128;
+	}
+	else if (row_size == 256) {
+		fw_meta_offset = HPI_META_DATA_OFFSET_ROW_256;
+	} else {
+		fw_meta_offset = 0;
+		g_set_error (error,
+			     FWUPD_ERROR,
+			     FWUPD_ERROR_NOT_SUPPORTED,
+			     "not support row size");
+		return FALSE;
+	}
+
+	device_data->fw_meta_offset = fw_meta_offset;
+	device_data->fw1_meta_row_num = row_max-1;
+	device_data->fw2_meta_row_num = row_max-2;
+
+	device_data->fw_meta_valid = FALSE;
+	if (!fu_ccgx_hpi_cmd_enter_flash_mode(device,hpi_handle,error)) {
+		return FALSE;
+	}
+
+	g_usleep (HPI_CMD_ENTER_FLASH_MODE_DELAY_US);
+
+	/* fw1 meta data */
+	fw_index = FW_MODE_FW1;
+	fw_meta_row_num = device_data->fw1_meta_row_num;
+
+	(void)fu_ccgx_hpi_clear_all_event (device, hpi_handle, 10, error);
+
+	if (!fu_ccgx_hpi_cmd_read_flash (device, hpi_handle, fw_meta_row_num, row_buffer, row_size,error)) {
+		return FALSE;
+	}
```

Comment:
I'd say no need for curly braces on this one if you just have one line (return FALSE)

Reply:
OK. I will remove braces for one line if. I have a automotive MISRA-C coding style. :)

Reply:
@superm1 once tried to codify my pet-whitespace standard. It was impossible, so don't worry too much if you have to be reminded :)